src/katsdpmodels/primary_beam.py

```diff
@@ -0,0 +1,256 @@
+################################################################################
+# Copyright (c) 2021, National Research Foundation (SARAO)
+#
+# Licensed under the BSD 3-Clause License (the "License"); you may not use
+# this file except in compliance with the License. You may obtain a copy
+# of the License at
+#
+#   https://opensource.org/licenses/BSD-3-Clause
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+"""Primary beam models."""
+
+import enum
+from typing import Sequence, Tuple, Dict, ClassVar, Union, Any
+from typing_extensions import Literal
+
+import numpy as np
+try:
+    from numpy.typing import ArrayLike
+except ImportError:
+    ArrayLike = Any  # type: ignore
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+import h5py
+
+from . import models
+
+
+class AltAzFrame:
+    """Coordinate system aligned with the antenna.
+
+    The l coordinate is horizontal and increases with increasing azimuth (north
+    to east), while the m coordinate is vertical and increases with increasing
+    altitude. Both are defined by an orthographic (SIN) projection, with the
+    nominal pointing centre at zero.
+    """
+
+
+class RADecFrame:
+    """Coordinate system aligned with the celestial sphere.
```

Comment:
It could really be any RA/Dec frame as long as you supply a corresponding parallactic angle (I'm ignoring the distortions due to aberration since I think they're probably much smaller than the details in the beam).

Reply:
Maybe I should change "the celestial sphere" to "a celestial sphere", and clarify it in the details?